src/FakeItEasy/RootModule.cs

```diff
@@ -106,7 +106,7 @@ public static void RegisterDependencies(DictionaryContainer container)
 
             container.RegisterSingleton(c => new EventHandlerArgumentProviderMap());
 
-            container.Register(c => new SequentialCallContext(c.Resolve<CallWriter>(), c.Resolve<StringBuilderOutputWriter.Factory>()));
+            container.RegisterSingleton<SequentialCallContext.Factory>(c => () => new SequentialCallContext(c.Resolve<CallWriter>(), c.Resolve<StringBuilderOutputWriter.Factory>()));

```

Comment:
Small optimization: resolve the `CallWriter` and `StringBuilderOutputWriter.Factory` outside of the inner delegate, i.e.

```csharp
container.RegisterSingleton<SequentialCallContext.Factory>(c =>
{
    var callWriter = c.Resolve<CallWriter>();
    var outputWriterFactory = c.Resolve<StringBuilderOutputWriter.Factory>();
    return () => new SequentialCallContext(callWriter, outputWriterFactory);
});
```

This could probably be done in other places, too. Not sure how big the performance impact would be, but I guess it can't hurt.

Reply:
In fact, ignore this comment. I had forgotten that it was a singleton, so it will only be constructed once anyway.

Reply:
Also, those changes will come in time!